overrides/build-systems.json

```diff
@@ -833,6 +833,10 @@
   "pixelmatch": [
     "poetry-core"
   ],
+  "platformdirs": [
+    "hatchling",
+    "packaging"
```

Comment:
This file only contains the build system used. So it's good place to list hatchling for platformdirs, but not this kind of stuff.

I was thinking something more like adding to `overrides/default.nix`:

```nix
hatchling = super.hatchling.overridePythonAttrs (
  old: {
    propagatedBuildInputs = (old.propagatedBuildInputs or [ ]) ++ [
      self.editables
      self.packaging
      self.pathspec
      self.pluggy
      self.tomli
    ] ++ lib.optionals (self.pythonOlder "3.8") [
      self.importlib-metadata
    ];
  }
);
```